kola/harness.go

```diff
@@ -20,53 +20,57 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/coreos/mantle/cli"
-	"github.com/coreos/mantle/cmd/kola/etcdtests"
 	"github.com/coreos/mantle/platform"
 )
 
-var cmdEtcd = &cli.Command{
-	Run:     runEtcd,
-	Name:    "etcd",
-	Summary: "Run etcd cluster under QEMU (requires root)",
-	Description: `Run and kill etcd cluster
+// tests should register by adding init() functions to this package
+var Tests []Test
 
-Work in progress: the code sets up an etcd cluster and then calls
-integration tests to see how etcd interacts with new CoreOS images.
-Currently, a local discovery service is available through an embedded
-etcd server.
-
-This must run as root!
-`}
-
-func init() {
-	cli.Register(cmdEtcd)
+type Test struct {
+	Run         func(platform.Cluster) error
+	Name        string
+	Group       string // used to manually group related tests
```

Comment:
Lets do globbing, that is easier to do on command line than regex. So kola run 'Etcd*' to do etcd tests. Also lets make test names start with uppercase. Later if we add tags like 'long' or 'short' they can be lower case so kola run short isn't ambiguous.